specutils/core/generic.py

```diff
@@ -0,0 +1,119 @@
+from specutils.spectrum1d import Spectrum1D
+
+import numpy as np
+import logging
+
+from astropy.nddata import NDArithmeticMixin, NDSlicingMixin, NDIOMixin
+import astropy.units as u
+from astropy.wcs import WCS
+
+
+class GenericSpectrum1D(NDIOMixin, NDSlicingMixin, NDArithmeticMixin,
+                        Spectrum1D):
```

Comment:
This name seems fine to me. I don't think there is any prescription for how things that subclass from `NDData` should be named, and **not** incorporating `Data` in the name honestly seems better at this point.